apps/financial-aid/backend/src/app/modules/application/test/applicationControllerGuards.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { IdsUserGuard } from '@island.is/auth-nest-tools'
+import { CanActivate } from '@nestjs/common'
+import { RolesGuard } from '../../../guards/roles.guard'
+
+import { ApplicationController } from '../application.controller'
+
+describe('ApplicationController - guards', () => {
+  let guards: any[]
+
+  beforeEach(() => {
+    guards = Reflect.getMetadata('__guards__', ApplicationController)
+  })
+
+  it('should have one guards', () => {
+    expect(guards).toHaveLength(1)
+  })
+
+  describe('IdsUserGuard', () => {
+    let guard: CanActivate
+
+    beforeEach(() => {
+      guard = new guards[0]()
+    })
+
+    it('should have IdsUserGuard as quard 1', () => {
+      expect(guard).toBeInstanceOf(IdsUserGuard)
+    })
+  })
+})
+
+describe('ApplicationController - Checks if user has a current application for this period guards', () => {
+  let guards: any[]
+
+  beforeEach(() => {
+    guards = Reflect.getMetadata(
+      '__guards__',
+      ApplicationController.prototype.getCurrentApplication,
+    )
+  })
+
+  it('should have one guards', () => {
+    expect(guards).toHaveLength(1)
+  })
+
+  describe('RolesGuard', () => {
+    let guard: CanActivate
+
+    beforeEach(() => {
+      guard = new guards[0]()
+    })
+
+    it('should have RolesGuard as quard 1', () => {
+      expect(guard).toBeInstanceOf(RolesGuard)
+    })
+  })
+})
+
+describe('ApplicationController - Searches for application by nationalId guards', () => {
+  let guards: any[]
+
+  beforeEach(() => {
+    guards = Reflect.getMetadata(
+      '__guards__',
+      ApplicationController.prototype.findApplication,
+    )
+  })
+
+  it('should have one guards', () => {
+    expect(guards).toHaveLength(2)
+  })
+
+  describe('RolesGuard', () => {
+    let guard: CanActivate
+
+    beforeEach(() => {
+      guard = new guards[0]()
+    })
+
+    it('should have RolesGuard as quard 1', () => {
+      expect(guard).toBeInstanceOf(RolesGuard)
+    })
+  })
```

Comment:
Is a test missing here for the second guard?

Reply:
yess it needs module to work, so I decided to have it in another PR